### PR TITLE
[CARBONDATA-3561] Fix incorrect results after execute delete/update operation if there are null values

### PR DESCRIPTION
**Pull request by zzcclp**
**Problem:**
If there are null values in table, after execute delete/update sql, the results are incorrect when query with vector mode

**Root cause:**
In method 'fillVector' of 'LocalDictDimensionDataChunkStore', if the type of 'vector' is 'ColumnarVectorWrapperDirectWithDeleteDelta', the parameter 'counter' of 'ColumnarVectorWrapperDirectWithDeleteDelta' does not increase when the data is not null, so it will set the null value to the wrong index.

**Solution:**
Increase 'counter' of 'ColumnarVectorWrapperDirectWithDeleteDelta' when the data is not null.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?  No
 
 - [ ] Any backward compatibility impacted?  No
 
 - [ ] Document update required?  No

 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

File: datamap/examples/src/minmaxdatamap/main/java/org/apache/carbondata/datamap/examples/MinMaxIndexDataMapFactory.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ public DataMapBuilder createBuilder(Segment segment, String shardName,
    * @param segment
    */
   @Override
-  public void clear(Segment segment) {
+  public void clear(String segmentNo) {
```
Comment (zzcclp) [inline datamap/examples/src/minmaxdatamap/main/java/org/apache/carbondata/datamap/examples/MinMaxIndexDataMapFactory.java]: These changes are not related to this  issue, just fix  some compile errors after PR#3434.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/813/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/817/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/820/

**Comment by zzcclp** [comment]
@jackylk @ravipesala this pr is ready, please help to review, thanks.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/818/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/821/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/824/

**Comment by zzcclp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/820/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/823/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/826/

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/DeleteCarbonTableTestCase.scala [hunk 2]
```diff
@@ -380,6 +382,38 @@ class DeleteCarbonTableTestCase extends QueryTest with BeforeAndAfterAll {
...
+    checkAnswer(sql(s"select count(1) from ${tableName} where b is null"), Seq(Row(4)))
+
+    checkAnswer(sql(s"delete from ${tableName} where b ='4'"), Seq(Row(1)))
+    checkAnswer(sql(s"delete from ${tableName} where a ='9'"), Seq(Row(1)))
+    checkAnswer(sql(s"update ${tableName} set (b) = ('10') where a = '10'"), Seq(Row(1)))
+
+    checkAnswer(sql(s"select count(1) from ${tableName} where b is null"), Seq(Row(2)))
+    checkAnswer(sql(s"select * from ${tableName} where a = '1'"), Seq(Row("1", "1", "2017")))
+    checkAnswer(sql(s"select * from ${tableName} where a = '10'"), Seq(Row("10", "10", "2017")))
+
```
Comment (jackylk) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/DeleteCarbonTableTestCase.scala]: add one more `sql(s"select count(1) from ${tableName} where b is null")`

**Comment by jackylk** [comment]
LGTM

**Comment by zzcclp** [comment]
This pr has been merged, close it.
